core/evm.go

```diff
@@ -96,6 +101,14 @@ func newEVMBlockContext(header *types.Header, chain ChainContext, author *common
 	if header.ExcessBlobGas != nil {
 		blobBaseFee = eip4844.CalcBlobFee(*header.ExcessBlobGas)
 	}
+
+	// Durango enables the Shanghai upgrade of eth, which takes place after the
+	// Merge upgrade.
+	isDurango := params.GetExtra(chain.Config()).IsDurango(header.Time)
```

Comment:
upstream avoids this dependency here (on chain config/rules) by checking if Difficulty is 0,
but we cannot rely on that because we already made blocks with difficulty set to 1.